### PR TITLE
Create Commander command for multisignatures - Closes #4905

### LINKED ISSUE #4905: Create Commander command for multisignatures
### Description

With the changes introduced by [LIP-0017](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0017.md) the multi-signature related commands need to be re-designed and re-implemented.

```
transaction:create -t 12 --mandatory-key="xxx" --mandatory-key="yyy" \
--optional-key="xxx" --optional-key="yyy" --number-of-signatures=4 \
--passphrase="" --member-passphrase="" --member-passphrase=""
```

This command should ask first sender passphrase, and passphrases for mandatory keys and optional keys should be asked.

### Motivation

[LIP-0017](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0017.md) changed the way multi-signatures and signing work so we need to design and restore this functionality

### Acceptance Criteria
- Only accept passphrase from flags and prompt
- if no passphrase are input from flag, it should ask from prompt
- Test case where passphrases are input from flag
- Test case where passphrases are input from prompt
- Test cases where number of signatures value is invalid
- Test case where there is no mandatory keys
- Test case where there is no optional keys
- Test case where there is empty passphrase

### Related Issues
#4836

### PR DESCRIPTION
**Pull request by nazarhussain**
### What was the problem?

This PR resolves #4905

### How was it solved?

Created new command for multisig account registration. 

### How was it tested?

Run unit tests for commander.

### CODE REVIEW

File: commander/src/commands/transaction/create/multisignature.ts [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+			multiple: true,
+			description:
+				'Number of signatures required to validate transactions from that account. Its a less secure way to sign transaction. See command description for details. ',
+		}),
+	};
+
+	static description = `
+	Creates a transaction which will register the account as a multisignature account if broadcast to the network, using the following arguments:
+
+		1. Mandatory keys to verify signatures for the multisignature account. Signatures for these kes must exists to verify every transaction from that account.
```
Comment (shuse2) [inline commander/src/commands/transaction/create/multisignature.ts]:
```suggestion
		1. Mandatory keys to verify signatures for the multisignature account. Signatures for these keys must exists to verify every transaction from that account.
```

File: commander/src/commands/transaction/create/multisignature.ts [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+		2. Optional keys to verify signatures for the multisignature account. Signatures for these kes can or can't exists to verify transaction from that account.
```
Comment (shuse2) [inline commander/src/commands/transaction/create/multisignature.ts]:
```suggestion
		2. Optional keys to verify signatures for the multisignature account. Signatures for these keys may or may not exists to verify transaction from that account.
```

File: commander/src/commands/transaction/create/multisignature.ts [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+				passphrase: passphraseSource,
+				'member-passphrase': memberPassphrasesSource,
+				'optional-key': optionalKeysSource,
+				'mandatory-key': mandatoryKeysSource,
+				'number-of-signatures': numberOfSignaturesSource,
+			},
+		} = this.parse(MultisignatureCommand);
+
+		const numberOfSignatures = numberOfSignaturesSource || 0;
+		const mandatoryKeys = mandatoryKeysSource || [];
```
Comment (shuse2) [inline commander/src/commands/transaction/create/multisignature.ts]: this is `[]` if it doesn't exist and same for below with `multiple: true`?

File: commander/src/commands/transaction/create/multisignature.ts [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+			description: 'Transaction fee in LSK.',
+		},
+	];
+
+	static flags = {
+		...BaseCommand.flags,
+		networkIdentifier: flagParser.string(commonFlags.networkIdentifier),
+		passphrase: flagParser.string(commonFlags.passphrase),
+		'number-of-signatures': flagParser.integer({
+			description:
```
Comment (shuse2) [inline commander/src/commands/transaction/create/multisignature.ts]: We can set this as required?

File: commander/src/commands/transaction/create/multisignature.ts [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+				'Number of signatures required to validate transactions from that account',
+		}),
+		'mandatory-key': flagParser.string({
+			multiple: true,
```
Comment (shuse2) [inline commander/src/commands/transaction/create/multisignature.ts]: same as above

File: commander/src/commands/transaction/create/multisignature.ts [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+			description: 'Mandatory public key for multisignature account.',
+		}),
+		'optional-key': flagParser.string({
+			multiple: true,
```
Comment (shuse2) [inline commander/src/commands/transaction/create/multisignature.ts]: same as above

File: commander/src/commands/transaction/create/multisignature.ts (hunk 3 above)
Comment (nazarhussain) [reply commander/src/commands/transaction/create/multisignature.ts]: No it appears `undefined` if non is provided.

File: commander/src/commands/transaction/create/multisignature.ts (hunk 5 above)
Comment (nazarhussain) [reply commander/src/commands/transaction/create/multisignature.ts]: Mandatory Keys are not required.

File: commander/src/commands/transaction/create/multisignature.ts (hunk 6 above)
Comment (nazarhussain) [reply commander/src/commands/transaction/create/multisignature.ts]: Optional keys are itself not required. Either optional or mandatory keys must be given, and that case is covered within lisk-element.

File: commander/src/commands/transaction/create/multisignature.ts (hunk 5 above)
Comment (shuse2) [reply commander/src/commands/transaction/create/multisignature.ts]: true, i got confused 🤔

File: commander/src/commands/transaction/create/multisignature.ts (hunk 4 above)
Comment (nazarhussain) [reply commander/src/commands/transaction/create/multisignature.ts]: I don't want it to put business logic on the command level. The existence check already there in `lisk-elements` and if this option is not provided proper error message is shown to user.

File: commander/src/commands/transaction/create/multisignature.ts [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+			description: 'Optional public key for multisignature account.',
+		}),
+		'member-passphrase': flagParser.string({
+			multiple: true,
+			description:
+				'Number of signatures required to validate transactions from that account. Its a less secure way to sign transaction. See command description for details. ',
```
[the remark this answers is not in the document]
Comment (nazarhussain) [reply commander/src/commands/transaction/create/multisignature.ts]: I don't get it.

[the remark this answers is not in the document]
Comment (ManuGowda) [reply commander/src/commands/transaction/create/multisignature.ts]: the flag is `member-passphrase` but the `description` doesn't correspond well with flag.

Comment (nazarhussain) [inline commander/src/commands/transaction/create/multisignature.ts]:
```suggestion
				'Passphrase for all mandatory and optional keys. Its a less secure way to sign transaction. See command description for details. ',
```

[the remark this answers is not in the document]
Comment (nazarhussain) [reply commander/src/commands/transaction/create/multisignature.ts]: Updated the description. Please check.

**Pull request merged**